2-Draft-Accepted/RFCxxxx-DSC-for-PowerShell7.md

```diff
@@ -0,0 +1,318 @@
+    "GenerationDate": "11/05/2020 11:09:47",
+    "MinimumCompatibleVersion": "1.0.0",
+    "CompatibleVersionAdditionalProperties": [
+      "BaseResource:ConfigurationName"
+    ],
+    "Name": "FileResourceDemo",
+    "Version": "2.0.0",
+    "Author": "slee",
+    "GenerationHost": "SLEE-DESKTOP"
+  },
+  "LogResource": [
+    {
+      "SourceInfo": "C:\\\\Users\\\\slee\\\\test\\\\file.dsc.ps1::15::9::Log",
+      "ModuleVersion": "1.0",
+      "Message": "Finished running the file resource with ID DirectoryCopy",
+      "ResourceID": "[Log]AfterDirectoryCopy",
+      "ModuleName": "PsDesiredStateConfiguration",
+      "ConfigurationName": "FileResourceDemo",
+      "DependsOn": [
+        "[File]DirectoryCopy"
+      ]
+    }
+  ],
+  "FileDirectoryConfiguration": [
+    {
+      "ResourceID": "[File]DirectoryCopy",
+      "Ensure": "Present",
+      "Recurse": true,
+      "ModuleVersion": "1.0",
+      "DestinationPath": "C:\\\\Users\\\\Public\\\\Documents\\\\DSCDemo\\\\DemoDestination",
+      "ConfigurationName": "FileResourceDemo",
+      "SourceInfo": "C:\\\\Users\\\\slee\\\\test\\\\file.dsc.ps1::6::9::File",
+      "ModuleName": "PSDesiredStateConfiguration",
+      "Type": "Directory",
+      "SourcePath": "\\\\\\\\PullServer\\\\DemoSource"
+    }
+  ]
+}
+```
+
+Important differences:
+
+- CIM class name prefixes are removed as they are not needed
+- Instances are represented as an array of the type name
+- The comment at the start is dropped as that information is available within `ConfigurationDocument`
+
+### No longer require MOF schema by switching to improved class-based resources
+
+Although DSC resources can be written as WMI/OMI providers, this requires writing
+C/C++ code and compiling for specific Linux distros and processor architectures.
+Writing PowerShell script DSC resources currently still requires a matching schema
+file making it more complex. Instead, the current plan is to
+[only support PowerShell class based DSC resources](https://github.com/PowerShell/PowerShell/issues/13731)
+in DSC for PowerShell 7. Whereas previously embedded objects were CIMInstance types,
+they would now be .NET types based on PowerShell classes. Additional reasoning
+and discussion is in the issue linked above.
+
+Making the change from script based resources to class based resources could make
+learning DSC a frustrating experience for new users. To resolve this, tooling will
+be needed that scaffolds as much as possible of the class requirements based
+on a .psm1 module file containing traditionally authored Get/Set/Test-TargetResource
+functions. This would also aid in migrating the more than 1000 existing resources
+for Windows.
+
+### Open Source
+
+The current
+[PSDesiredStateConfiguration](https://docs.microsoft.com/powershell/module/psdesiredstateconfiguration/?view=powershell-7)
+module is not Open Source. The intent is to clean up the code and prepare the GitHub
+repository to be made public as an Open Source project.
```

Comment:
"open source" is usually lower-case, but it's up to you here.

Also might be good to include a non-ETA "though we don't currently have a date for this work" or whatever